src/library/handlers/traceHandler.ts

```diff
@@ -67,16 +80,41 @@ export class TraceHandler {
         // TODO: Check for conflicts with multiple handlers available
         this.tracer = this.tracerProvider.getTracer("ApplicationInsightsTracer");
 
+        // Defautl configs
         this.httpInstrumentationConfig = {
             ignoreOutgoingRequestHook: this._ignoreOutgoingRequestHook.bind(this),
             ignoreIncomingRequestHook: this._ignoreIncomingRequestHook.bind(this),
         };
+        this.mongoDbInstrumentationConfig = {};
+        this.mySqlInstrumentationConfig = {};
+        this.azureSdkInstrumentationConfig = {};
+        this.redisInstrumentationConfig = {};
+        this.redis4InstrumentationConfig = {};
     }
 
     public start() {
         if (this._config.enableAutoCollectRequests || this._config.enableAutoCollectDependencies) {
-            let httpInstrumentation = new HttpInstrumentation(this.httpInstrumentationConfig);
-            this.addInstrumentation(httpInstrumentation);
+            this.addInstrumentation(new HttpInstrumentation(this.httpInstrumentationConfig));
+        }
+        if (this._config.instrumentations) {
+            if (this._config.instrumentations[InstrumentationType.azureSdk] && this._config.instrumentations[InstrumentationType.azureSdk].enabled) {
+                this.addInstrumentation((createAzureSdkInstrumentation(this.azureSdkInstrumentationConfig)) as any);
+            }
+            if (this._config.instrumentations[InstrumentationType.mongoDb] && this._config.instrumentations[InstrumentationType.mongoDb].enabled) {
```

Comment:
Suggest: This can perhaps be moved to a function like so: private isInstumentationEnabled(string instrumentationType):boolean {}